x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/predicate/conditional/CaseTests.java

```diff
@@ -0,0 +1,53 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+package org.elasticsearch.xpack.sql.expression.predicate.conditional;
+
+import org.elasticsearch.xpack.sql.expression.Expression;
+import org.elasticsearch.xpack.sql.expression.predicate.operator.comparison.Equals;
+import org.elasticsearch.xpack.sql.tree.AbstractNodeTestCase;
+import org.elasticsearch.xpack.sql.tree.NodeSubclassTests;
+import org.elasticsearch.xpack.sql.tree.Source;
+
+import java.util.Arrays;
+
+import static org.elasticsearch.xpack.sql.expression.function.scalar.FunctionTestUtils.randomIntLiteral;
+import static org.elasticsearch.xpack.sql.expression.function.scalar.FunctionTestUtils.randomStringLiteral;
+
+/**
+ * Needed to override tests in {@link NodeSubclassTests} as Case is special since its children are not usual
+ * expressions but {@link IfConditional}s.
+ */
+public class CaseTests extends AbstractNodeTestCase<Case, Expression> {
+
+    public static Case randomCase() {
+        return new Case(Source.EMPTY, Arrays.asList(
+            new IfConditional(Source.EMPTY, new Equals(Source.EMPTY, randomStringLiteral(), randomStringLiteral()),
+                randomIntLiteral()), randomIntLiteral()));
+    }
+
+    @Override
+    protected Case randomInstance() {
+        return randomCase();
+    }
+
+    @Override
+    protected Case mutate(Case instance) {
+        return randomCase();
```

Comment:
I don't think this `mutate` logic will always have a changed, non-equal to original, instance of CASE. I think the mutation should have as many `IfConditional` instances as constructor values that can change (2 in this case with `randomStringLiteral()`). And from `Case` point of view, it will also need to mutate the `randomIntLiteral()` parameter as well. Something like this: https://github.com/elastic/elasticsearch/blob/master/x-pack/plugin/sql/src/test/java/org/elasticsearch/xpack/sql/expression/function/scalar/string/ConcatFunctionPipeTests.java#L96